### ISSUE TITLE
ErrorException : Undefined property: stdClass::$type

### ISSUE THREAD

**Issue by xbojch** [hotmeteor/spectator#38]
The `allOf` property is not resolved correctly and doesn't act as a replacement for the `type` property.
Validating a response with a spec below (only the relevant parts) results in the error in the issue title.

Is this an issue in Spectator or related to https://github.com/cebe/php-openapi/issues/31?

**Workaround**
Replacing the `allOf` with a `type` property resolves the error.

```json
{
  "components": {
    "schemas": {
      "Name": {
        "type": "object",
        "properties": {
          "name": {
            "type": "string"
          }
        }
      },
      "MySchema": {
        "title": "MySchema",
        "allOf": [
          {
            "$ref": "#/components/schemas/Name"
          },
          {
            "type": "object",
            "properties": {
              "type": {
                "type": "integer"
              },
              "description": {
                "type": "string"
              }
            },
            "required": [
              "type"
            ]
          }
        ]
      }
    }
  }
}
```

**Comment by xbojch** [comment]
I ended up using https://github.com/osteel/openapi-httpfoundation-testing with https://github.com/DarkaOnLine/L5-Swagger as described [here](https://hackernoon.com/validate-your-php-api-tests-against-openapi-definitions-a-laravel-example).

**Comment by hotmeteor** [comment]
@xbojch Can you provide more information on what you're trying to do? I added a test for this and didn't hit an issue, but I'm sure I'm not replicating it properly: https://github.com/hotmeteor/spectator/blob/v1.0.0/tests/ResponseValidatorTest.php#L277

**Comment by xbojch** [comment]
The `Components.v1.yaml` looks correct. The path `/` that's being tested is not defined in it though or am I missing something.
The schema`MySchema` seems to not be used anywhere. Try referencing the `MySchema` in a response, that tripped it for me.

**Comment by hotmeteor** [comment]
@xbojch This should be fixed in v1.0.1. There is an issue with the parsing of component references, so I've got a workaround until the underlying package is fixed.

**Comment by xbojch** [comment]
Thank you very much @hotmeteor. I'll give it another try, because I just love how much shorter and more understandable it makes the tests.
